pkg/rotator/rotator.go

```diff
@@ -578,67 +602,35 @@ func (r *ReconcileWH) Reconcile(request reconcile.Request) (reconcile.Result, er
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileWH) ensureVWHCerts(certPem []byte) error {
-	vwh := &unstructured.Unstructured{}
-	vwh.SetGroupVersionKind(vwhGVK)
-	if err := r.client.Get(r.ctx, r.vwhKey, vwh); err != nil {
-		if k8sErrors.IsNotFound(err) {
-			crLog.Info("VWK " + r.vwhKey.Name + " is not found. No action is needed.")
-			return nil
-		}
-		// Error reading the object - requeue the request.
-		return err
-	}
-
-	crLog.Info("ensuring CA cert on ValidatingWebhookConfiguration")
-	if err := injectCertToWebhook(vwh, certPem); err != nil {
-		crLog.Error(err, "unable to inject cert to webhook")
-		return err
-	}
-	if err := r.client.Update(r.ctx, vwh); err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// ensureCRDConvWHCerts returns an arbitrary error if multiple errors are
-// encountered, while all the errors are logged.
-func (r *ReconcileWH) ensureCRDConvWHCerts(certPem []byte) error {
-	var anyError error = nil
-	for _, crdName := range r.crdNames {
-		crd := &unstructured.Unstructured{}
-		crd.SetGroupVersionKind(crdGVK)
-		crdKey := types.NamespacedName{Name: crdName}
-		if err := r.client.Get(r.ctx, crdKey, crd); err != nil {
+func (r *ReconcileWH) ensureCerts(certPem []byte) error {
+	for _, webhook := range r.webhooks {
+		updatedResource := &unstructured.Unstructured{}
+		gvk := typeToGVKMap[webhook.Type]
+		updatedResource.SetGroupVersionKind(gvk)
+		if err := r.client.Get(r.ctx, types.NamespacedName{Name: webhook.Name}, updatedResource); err != nil {
 			if k8sErrors.IsNotFound(err) {
-				crLog.Info("CRD " + crdName + " is not found")
-				continue
+				crLog.Info("Webhook not found. No action is needed.", "name", webhook.Name, "gvk", gvk)
```

Comment:
Hmm, I'm rethinking this behaviour. I think maybe we should _log_ it as an error ("you asked me to update this thing, but this thing didn't exist") but maybe not _return_ an error, since you only return an error to ask `controller-runtime` to try calling `Reconcile` again, and that won't help if the resource doesn't exist (if the resource is _created_, the reconciler will be called automatically).

So maybe change this to:

```
crLog.Error(err, "Webhook not found", "name", webhook.Name, "gvk", gvk)
continue
```

Also, a little tip about logs. You're using exactly the same two params in this whole function, and there's an easier way to do that:

```
log := crLog.WithValues("name", webhook.Name, "gvk", gvk)
...
log.Error(err, "Webhook not found") // automatically adds everything from WithValues()
```

Up to you if you think that's worth it, it's just a neat logging trick.

Reply:
Thanks for the tip.
Updated.